store/store.go

```diff
@@ -69,6 +70,19 @@ const (
 	UbuntuCoreWireProtocol = "1"
 )
 
+type rateLimitKey string
```

Comment:
Not a blocker, but it's common to use just a struct{} for that purpose. The type itself becomes the key.

Reply:
Yeah, indeed. Thanks for this! Fwiw, if we add a "store.DownloadOptions" we don't need this context anymore. It will add a bit of churn to the PR because of test updates though: http://paste.ubuntu.com/p/7kvbVq4ZBD/